app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/DisclosePersonalDetailsConsentController.scala

```diff
@@ -55,12 +56,48 @@ class DisclosePersonalDetailsConsentController @Inject() (
               isInReviewMode = false,
               disclosePersonalDetailsYesNoAnswerForm,
               requestSessionData,
-              service
+              service,
+              textPara2(requestSessionData.userSelectedOrganisationType),
+              questionLabel(requestSessionData.userSelectedOrganisationType)
             )
           )
         )
     }
 
+  private def textPara2(cdsOrganisationType: Option[CdsOrganisationType])(implicit messages: Messages): String =
+    cdsOrganisationType match {
+      case orgType if orgType.contains(CdsOrganisationType.Company) =>
+        messages("ecc.subscription.organisation-disclose-personal-details-consent.org.para2")
+      case _
+          if cdsOrganisationType.contains(CdsOrganisationType.Partnership) || cdsOrganisationType.contains(
+            CdsOrganisationType.LimitedLiabilityPartnership
+          ) =>
+        messages("ecc.subscription.organisation-disclose-personal-details-consent.partnership.para2")
+      case _
+          if cdsOrganisationType.contains(
+            CdsOrganisationType.CharityPublicBodyNotForProfit
+          ) || cdsOrganisationType.contains(CdsOrganisationType.ThirdCountryOrganisation) =>
+        messages("ecc.subscription.organisation-disclose-personal-details-consent.charity.para2")
+      case _ => messages("ecc.subscription.organisation-disclose-personal-details-consent.individual.para2")
+    }
+
+  private def questionLabel(cdsOrganisationType: Option[CdsOrganisationType])(implicit messages: Messages): String =
+    cdsOrganisationType match {
+      case orgType if orgType.contains(CdsOrganisationType.Company) =>
+        messages("ecc.subscription.organisation-disclose-personal-details-consent.org.question")
+      case _
+          if cdsOrganisationType.contains(CdsOrganisationType.Partnership) || cdsOrganisationType.contains(
+            CdsOrganisationType.LimitedLiabilityPartnership
+          ) =>
+        messages("ecc.subscription.organisation-disclose-personal-details-consent.partnership.question")
+      case _
+          if cdsOrganisationType.contains(
+            CdsOrganisationType.CharityPublicBodyNotForProfit
+          ) || cdsOrganisationType.contains(CdsOrganisationType.ThirdCountryOrganisation) =>
+        messages("ecc.subscription.organisation-disclose-personal-details-consent.charity.question")
+      case _ => messages("ecc.subscription.organisation-disclose-personal-details-consent.individual.question")
+    }
+
```

Comment:
I think this may look a bit tidier if we defaulted what was passed in before the pattern matching:

```
    val org = cdsOrganisationType.getOrElse(CdsOrganisationType.Individual)
    org match {
      case CdsOrganisationType.Company =>
        messages("ecc.subscription.organisation-disclose-personal-details-consent.org.question")
      case CdsOrganisationType.Partnership) || CdsOrganisationType.LimitedLiabilityPartnership =>
        messages("ecc.subscription.organisation-disclose-personal-details-consent.partnership.question")
      case CdsOrganisationType.CharityPublicBodyNotForProfit || CdsOrganisationType.ThirdCountryOrganisation =>
        messages("ecc.subscription.organisation-disclose-personal-details-consent.charity.question")
      case _ => messages("ecc.subscription.organisation-disclose-personal-details-consent.individual.question")
    }
```

Reply:
I'd also suggest we relocate this logic to a ViewModel (maybe everything the view requires? Boy scout rule and all)